src/utils/logger.ts

```diff
@@ -0,0 +1,25 @@
+export default class Logger {
+  console: globalThis.Console | any;
+
+  name: string;
+
+  // If a console has been passed, use it. Otherwise use window.console
+  constructor(name?: string, console?: globalThis.Console | any) {
+    this.console = console ?? window.console;
+    this.name = name ?? 'ewc';
+  }
+
+  // Produce a namespaced console warning
+  warn(message: any) {
+    if (this.console && this.console.warn != null) {
+      this.console.warn(`${this.name}: ${message}`);
+    }
+  }
```

Comment:
I'm not sure this will work with `any` input type. Potentially the following would be a better fit:

```suggestion
  warn(message: any) {
    if (typeof this.console?.warn !== "function") {
      // Handle this case somehow
    }
  
    if (["string", "number", "boolean"].includes(typeof message)) { 
      this.console.warn(`${this.name}: ${message}`);
    } else {
      this.console.warn(`${this.name}: `, message);
    }
  }
```

You could also extract this function further to reuse it between all of the different log types fairly easily. If you are going to build a logger, it should expose the same functionality as the Console (ie. `assert()`, `clear()`, `count()`, etc).